### PR TITLE
feat: cds.Vector support for the HANAService

### PR DESCRIPTION
**Pull request by BobdenOs**

### CODE REVIEW

File: hana/lib/HANAService.js [hunk 1]
```diff
@@ -908,6 +909,7 @@ class HANAService extends SQLService {
       // Not encoded string with CESU-8 or some UTF-8 except a surrogate pair at "base64_decode" function
       Binary: e => `HEXTOBIN(${e})`,
       Boolean: e => `CASE WHEN ${e} = 'true' THEN TRUE WHEN ${e} = 'false' THEN FALSE END`,
+      Vector: e => `TO_REAL_VECTOR(${e})`,
```
Comment (David-Kunz) [inline hana/lib/HANAService.js:912]: We might also  send the binary variant over the wire (less bytes), but I'm fine to just use `TO_REAL_VECTOR` for now.

**Review by David-Kunz — APPROVED** [review]

**Comment by David-Kunz** [comment]
Hi @BobdenOs ,

Did you also do it for the old HANA driver? I think that's what people are waiting for most.

File: test/compliance/resources/db/basic/literals.cds [hunk 2]
```diff
@@ -53,6 +53,13 @@ entity array {
...
 
 entity binaries {
-     binary : Binary;
-     largebinary : LargeBinary;
+    binary      : Binary;
+    largebinary : LargeBinary;
 }
+
+
+/* Exlcuded from the tests until fully supported
```
Comment (David-Kunz) [inline test/compliance/resources/db/basic/literals.cds:61]:
```suggestion
/* Excluded from the tests until fully supported
```

File: hana/lib/HANAService.js [hunk 3]
```diff
@@ -894,6 +894,7 @@ class HANAService extends SQLService {
       LargeBinary: () => `NVARCHAR(2147483647)`,
       Binary: () => `NVARCHAR(2147483647)`,
       array: () => `NVARCHAR(2147483647)`,
+      Vector: () => `NVARCHAR(2147483647)`,
 
       // Javascript types
```
Comment (David-Kunz) [inline hana/lib/HANAService.js:899]:
```suggestion
      // JavaScript types
```

**Comment by David-Kunz** [comment]
Hi @BobdenOs ,

Can we merge this for our colleagues to test?

**Review by johannes-vogel — APPROVED** [review]

**Pull request merged**